### ISSUE TITLE
AbsSecurityGroupUnrestrictedIngress not correctly handling aws_vpc_security_group_ingress_rule terraform resource

### ISSUE THREAD

**Issue by FISHMANPET** [bridgecrewio/checkov#6624]
**Describe the issue**
This is related to CKV_AWS_260 specifically, but all that does is call AbsSecurityGroupUnrestrictedIngress.py, and the issue lies there I think

Specifically, the rule is generating a false positive because it's not finding the key `referenced_security_group_id` in the `aws_vpc_security_group_ingress_rule` resource.
**Examples**
``` terraform
resource "aws_vpc_security_group_ingress_rule" "ecs-inbound" {
  security_group_id = aws_security_group.ecs-sg.id

  referenced_security_group_id = aws_security_group.alb-sg.id
  from_port                    = 80
  to_port                      = 80
  ip_protocol                  = "tcp"
  description                  = "Allow traffic from load balancer into ECS"
}
```
I would expect this to not generate a violation of CKV_AWS_260, but it does.
**Version (please complete the following information):**
This has happened to me both locally and in GitHub actions where the versions are slightly different.
 - 3.2.200
 - 3.2.216

**Additional context**
I believe the issue might lie [here](https://github.com/bridgecrewio/checkov/blob/56babf6ef1620d6a966b40c6453cb639cd6ff309/checkov/terraform/checks/resource/aws/AbsSecurityGroupUnrestrictedIngress.py#L107). I believe that would be the code path that's generating a failure for my rule.  The key `security_groups` is used when setting rules directly in an `aws_security_group` resource, and `source_security_group_id` is used when setting a rule in a `aws_security_group_rule` resource, but this `aws_vpc_security_group_ingress_rule` resource uses `referenced_security_group_id` and there's no check for that here. So in this case the rule definition has basically fallen through and defaulted to failing because it hasn't actually detected a full configuration.

**Comment by claque2000** [comment]
I got the same issue on Checkov version 3.2.367.

`Check: CKV_AWS_260: "Ensure no security groups allow ingress from 0.0.0.0:0 to port 80"
	FAILED for resource: aws_vpc_security_group_ingress_rule.xxxx_lb_to_xxxx_ec2
	File: /instance.tf:68-78
	Guide: https://docs.prismacloud.io/en/enterprise-edition/policy-reference/aws-policies/aws-networking-policies/ensure-aws-security-groups-do-not-allow-ingress-from-00000-to-port-80

		68 | resource "aws_vpc_security_group_ingress_rule" "xxxx_lb_to_xxxx_ec2" {
		69 |   description                  = format("Allows communication from load_balancer security group %s", aws_security_group.xxx.id)
		70 |   security_group_id            = aws_security_group.yyy.id
		71 |   referenced_security_group_id = aws_security_group.xxx.id
		72 |   from_port                    = 80
		73 |   to_port                      = 80
		74 |   ip_protocol                  = "tcp"
		75 |   tags = {
		76 |     Name = format("xxxxx", aws_security_group.xxx.id)
		77 |   }
		78 | }
`

The access on port 80 is restricted to a specific source security group and not available from 0.0.0.0/0. The source of the issue identified seems to be correct as it does not seems to check the correct configuration parameter for at least aws_vpc_security_group_ingress_rule
